python/lsst/obs/hsc/hscMapper.py

```diff
@@ -127,7 +139,12 @@ def map(self, datasetType, dataId, write=False):
         """
         copyId = dataId.copy()
         copyId.pop("flags", None)
-        return super(HscMapper, self).map(datasetType, copyId, write=write)
+        location = super(HscMapper, self).map(datasetType, copyId, write=write)
+
+        if 'flags' in dataId:
+            location.getAdditionalData().set('flags', dataId['flags'])
+
+        return location
```

Comment:
I'd change it, since it's in the summary line: that's what people will see first when they read the logs, and we don't want them to be confused.